File: backend/app/domain/entities/project.py
from dataclasses import dataclass, field
from datetime import date, datetime
from typing import Any


@dataclass
class LatestRunInfo:
    run_number: int | None = None
    status: str | None = None
    completed_at: datetime | None = None
    created_at: datetime | None = None


@dataclass
class Project:
    id: str
    name: str
    organisation_id: str | None = None
    client: str | None = None
    currency: str = "EUR"
    valuation_date: date | None = None
    reporting_language: str = "de"
    status: str = "draft"
    team_name: str | None = None
    created_by: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None
    metadata: dict[str, Any] = field(default_factory=dict)


@dataclass
class ProjectSummary:
    id: str
    name: str
    client: str | None
    asset_count: int
    status: str
    latest_run: LatestRunInfo | None
    team_name: str | None
    created_at: datetime | None
    updated_at: datetime | None


@dataclass
class DashboardStats:
    open_projects: int
    assets_within_scope: int
    awaiting_review: int
    reports_per_month: int


@dataclass
class ProjectListResult:
    items: list[ProjectSummary]
    total: int

File: frontend/src/types/project.ts
export type ProjectStatus = 'draft' | 'test' | 'ready' | 'active' | 'archived'

export type LatestRun = {
  runNumber: number | null
  status: string | null
  completedAt: string | null
  createdAt: string | null
}

export type ProjectSummary = {
  id: string
  projectName: string
  clientName: string | null
  assetCount: number
  status: ProjectStatus
  latestRun: LatestRun | null
  assignedTeam: string | null
  createdAt: string | null
  updatedAt: string | null
}

export type ProjectListResponse = {
  items: ProjectSummary[]
  total: number
  page: number
  pages: number
}

export type ProjectDetail = {
  id: string
  projectName: string
  clientName: string | null
  currency: string
  valuationDate: string | null
  reportingLanguage: string
  status: ProjectStatus
  assignedTeam: string | null
  createdBy: string | null
  createdAt: string | null
  updatedAt: string | null
  metadata: Record<string, unknown>
}

export type DashboardStats = {
  openProjects: number
  assetsWithinScope: number
  awaitingReview: number
  reportsPerMonth: number
}

export type CreateProjectPayload = {
  client: string
  projectName: string
  currency: string
  valuationDate: string
  reportingLanguage: string
}
